Sources/BranchSDK/BNCPreferenceHelper.m

```diff
@@ -985,7 +985,7 @@ - (NSMutableDictionary *)deserializePrefDictFromData:(NSData *)data {
 
         dict = [NSKeyedUnarchiver unarchivedObjectOfClasses:classes fromData:data error:&error];
         if (error) {
-            [[BranchLogger shared] logWarning:@"Failed to load preferences from storage."];
+            [[BranchLogger shared] logWarning:@"Failed to load preferences from storage." error:error];
```

Comment:
Just for confirmation, can the log function work without passing in the error parameter? This hasn't been widely adopted yet right?

Reply:
Should have called this out in the PR. The reason I didn't leave a version with and without NSError is that it breaks Swift. 

The bridge between ObjC and Swift converts those NSErrors into Exceptions. So if you have both versions, you'll end up with a definition that throws and exception and a definition that does not. Although not breaking, this is very confusing to devs as there's two conflicting definitions and related warnings.

Reply:
And although `BranchLogger` is a public class, it's really intended for our use in the SDK and any wrappers. The actual public log API for clients has not changed as that always had an NSError on it.

Reply:
I see so you'd have to use a different name for the wrapping function.